application/gqlgen/post.go

```diff
@@ -589,3 +589,76 @@ func (r *mutationResolver) RemovePotentialProvider(ctx context.Context, postID,
 
 	return &post, nil
 }
+
+func (r *mutationResolver) MarkRequestAsDelivered(ctx context.Context, postID string) (*models.Post, error) {
+	cUser := models.CurrentUser(ctx)
+
+	var post models.Post
+	if err := post.FindByUUID(postID); err != nil {
+		return nil, domain.ReportError(ctx, err, "MarkRequestAsDelivered.FindPost")
+	}
+
+	extras := map[string]interface{}{
+		"user": cUser.UUID,
+		"post": post.UUID,
+	}
+
+	if cUser.AdminRole != models.UserAdminRoleSuperAdmin && cUser.ID != post.ProviderID.Int {
```

Comment:
Can this be done with `post.canUserChangeStatus` to keep all the logic on one place? (Same on the other mutation.)

Reply:
:+1: